src/Mvc/Mvc.Core/src/Infrastructure/SystemTextJsonResultExecutor.cs

```diff
@@ -123,11 +134,22 @@ private static class Log
                 new EventId(1, "JsonResultExecuting"),
                 "Executing JsonResult, writing value of type '{Type}'.");
 
+            private static readonly Action<ILogger, string, Exception> _eagerlyReadingAsyncEnumerable = LoggerMessage.Define<string>(
+               LogLevel.Debug,
+               new EventId(2, "EagerReadAsyncEnumerable"),
+               "Eagerly reading IAsyncEnumerable instance of type '{Type}'.");
```

Comment:
Can we say something like `Eagerly buffering` instead of reading? I think it's a little confusing to say that we're reading when we're about to write. I realize that it's technically correct, but still a little misleading.